src/sbl_filing_api/routers/filing.py

```diff
@@ -60,14 +58,16 @@ async def post_filing(request: Request, lei: str, period_code: str):
         try:
             return await repo.create_new_filing(request.state.db_session, lei, period_code)
         except IntegrityError:
-            raise HTTPException(
+            raise RegTechHttpException(
                 status_code=status.HTTP_409_CONFLICT,
+                name="Filing Creation Conflict",
                 detail=f"Filing already exists for Filing Period {period_code} and LEI {lei}",
             )
     else:
-        return JSONResponse(
-            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
-            content=f"The period ({period_code}) does not exist, therefore a Filing can not be created for this period.",
+        raise RegTechHttpException(
+            status_code=status.HTTP_404_NOT_FOUND,
```

Comment:
@billhimmelsbach @shindigira @meissadia not sure how far along front end is with the status code; is it ok to switch most of the 422s in the filing API which are filing doesn't exist, submission doesn't exist, or filing period doesn't exist to 404s? It makes more sense in my mind; since in this example a filing period doesn't exist, therefore a not found error.

Reply:
I am going to take your updates here and build a wiki page with response codes based on when and we can discuss those during devchat?